shell/components/novnc/NovncConsoleItem.vue

```diff
@@ -0,0 +1,89 @@
+<script>
+export default {
+  name: 'NovncConsoleItem',
+
+  props: {
+    items: {
+      type:     Object,
+      required: true,
+      default:  () => {
+        return {};
+      }
+    },
+
+    path: {
+      type:     Array,
+      required: true,
+      default:  () => {
+        return [];
+      }
+    },
+
+    pos: {
+      type:     Number,
+      required: true,
+      default:  0,
+    }
+  },
+
+  methods: {
+    keysDown(key, pos) {
+      this.addKeys({ key, pos });
+      this.$emit('sendKeys');
+    },
+
+    addKeys({ key, pos }) {
+      this.$emit('update', { key, pos });
+    },
+
+    sendKeys() {
+      this.$emit('sendKeys');
+    },
+
+    getOpenStatus(key, pos) {
+      return this.path[pos] === key;
```

Comment:
We should not directly change the value passed in by props.

Reply:
@WuJun2016 

Func `getOpenStatus` only return comparison result. The prop won't be modified.